### DISCUSSION TITLE
$default-branch not working

### DISCUSSION THREAD

**Discussion by mellis481** [community/community#26597] [Actions]
<p>I read in <a href="https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/" rel="noopener nofollow ugc">this</a> article that you can use <code>$default-branch</code> to reference your repo’s default branch instead of having to hardcode <code>main</code>/<code>master</code>/etc.  It isn’t working for me, however.  Here is how I was using <code>$default-branch</code> as a trigger:</p>
<pre><code class="lang-auto">on:
  push:
    branches: [$default-branch]
</code></pre>
<p>When I merged a feature branch PR into my base branch, the workflow whose trigger I defined above never got kicked off.  It was only when I changed the trigger to the following would it pick it up:</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - master
</code></pre>
<p>What am I doing wrong?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The documentation on <a href="https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization#creating-a-workflow-template">custom workflow templates</a> linked in that changelog post says this:</p>
<blockquote>
<p>If you need to refer to a repository’s default branch, you can use the <code>$default-branch</code> placeholder. When a workflow is created using your template, the placeholder will be automatically replaced with the name of the repository’s default branch.</p>
</blockquote>
<p>So <code>$default-branch</code> doesn’t seem to be supported in the actual workflow file.</p>

**Comment by mellis481** [comment]
<p>Ah. I clearly misunderstood the intent. Thanks for clearing that up.</p>
